main.js

```diff
@@ -5,54 +5,91 @@ const { spawn } = require('child_process')
 
 const appPath = app.getAppPath()
 const ignoredPaths = /node_modules|[/\\]\./
-// Main file poses a special case, as its changes are
-// only effective when the process is restarted (hard reset)
-// We assume that electron-reload is required by the main
-// file of the electron application
-const mainFile = module.parent.filename
 
 /**
  * Creates a callback for hard resets.
  *
- * @param {String} eXecutable path to electron executable
- * @param {String} hardResetMethod method to restart electron
+ * @param {string} eXecutable path to electron executable
+ * @param {string} hardResetMethod method to restart electron
+ * @param {string[]} argv arguments to restart electron with
  * @returns {Function} handler to pass to chokidar
  */
-const createHardresetHandler = (eXecutable, hardResetMethod, argv) =>
-  () => {
-    // Detaching child is useful when in Windows to let child
-    // live after the parent is killed
-    const args = (argv || []).concat([appPath])
-    const child = spawn(eXecutable, args, {
-      detached: true,
-      stdio: 'inherit'
-    })
-    child.unref()
-    // Kamikaze!
+const createHardresetHandler = (eXecutable, hardResetMethod, argv) => () => {
+  // Detaching child is useful when in Windows to let child
+  // live after the parent is killed
+  const args = [appPath].concat(argv)
+  const child = spawn(eXecutable, args, {
+    detached: true,
+    stdio: 'inherit'
+  })
+  child.unref()
+  // Kamikaze!
 
-    // In cases where an app overrides the default closing or quiting actions
-    // firing an `app.quit()` may not actually quit the app. In these cases
-    // you can use `app.exit()` to gracefully close the app.
-    if (hardResetMethod === 'exit') {
-      app.exit()
-    } else {
-      app.quit()
-    }
+  // In cases where an app overrides the default closing or quiting actions
+  // firing an `app.quit()` may not actually quit the app. In these cases
+  // you can use `app.exit()` to gracefully close the app.
+  if (hardResetMethod === 'exit') {
+    app.exit()
+  } else {
+    app.quit()
   }
+}
 
-module.exports = (glob, options = {}) => {
+/**
+ * @param {string | string[]} glob
+ * @param {Partial<chokidar.WatchOptions> & {@property {string} [hardResetMethod] @property {string[]} [argv] @property {boolean} [forceHardReset] @property {string} electron @property {string} [mainFile]}} [options]
```

Comment:
This is not a valid JSDoc [parameter type](https://jsdoc.app/tags-param.html):

> The parameter type can be a built-in JavaScript type, such as string or Object, or a JSDoc namepath to another symbol in your code. If you have written documentation for the symbol at that namepath, JSDoc will automatically link to the documentation for that symbol. You can also use a type expression to indicate, for example, that a parameter is not nullable or can accept any type; see the @type tag documentation for details.

Reply:
JSDoc types have been updated